src/Platforms/CUDA/AccelBLAS_CUDA.hpp

```diff
@@ -9,36 +9,38 @@
 // File refactored from: MatrixDelayedUpdateCUDA.h
 //////////////////////////////////////////////////////////////////////////////////////
 
-#ifndef QMCPLUSPLUS_CUDA_LINEAR_ALGEBRA_HANDLES_H
-#define QMCPLUSPLUS_CUDA_LINEAR_ALGEBRA_HANDLES_H
+#ifndef QMCPLUSPLUS_CUDA_ACCELBLAS_CUDA_H
+#define QMCPLUSPLUS_CUDA_ACCELBLAS_CUDA_H
 
-#include "ResourceCollection.h"
+#include "AccelBLAS.hpp"
 #include "CUDA/CUDAruntime.hpp"
-#include "CUDA/cuBLAS.hpp"
 #include "CUDA/QueueCUDA.hpp"
+#include "CUDA/cuBLAS.hpp"
 
 namespace qmcplusplus
 {
-struct CUDALinearAlgebraHandles : public Resource
+namespace compute
+{
+template<>
+class BLASHandle<PlatformKind::CUDA>
 {
-  // CUDA specific variables
-  compute::Queue<PlatformKind::CUDA> queue;
+  public:
+  // cuda stream, not owned, reference-only
+  cudaStream_t h_stream;
```

Comment:
the type is actually a pointer, the semantics are very handle like.  From an English standpoint your might say it refers to a stream but in C++ context I wouldn't really call it a reference.

Reply:
Not C++ reference type but conceptually a reference.